database/buildSchema/29_review_response.sql

```diff
@@ -11,7 +11,28 @@ CREATE TABLE public.review_response (
 	review_question_assignment_id integer references public.review_question_assignment(id),
 	application_response_id integer references public.application_response(id),
 	review_response_link_id integer references public.review_response(id),
+	original_response_id integer references public.review_response(id),
 	review_id integer references public.review(id),
     time_created timestamptz default current_timestamp,
 	status public.review_response_status default 'Draft'
-);
\ No newline at end of file
+);
+
+-- set review response original_response_id (the response that links to application id should be available for all reaponses)
+
+CREATE OR REPLACE FUNCTION set_original_response()
+RETURNS trigger AS $$
+BEGIN
+	IF NEW.application_response_id IS NOT NULL THEN
+	  NEW.original_response_id = NEW.id;
+	END IF;
+	IF NEW.review_response_link_id IS NOT NULL THEN
+	   NEW.original_response_id = (SELECT original_response_id FROM review_response where id = NEW.review_response_link_id);
+	END IF;
```

Comment:
It shouldn't be possible for application_response_id to be NULL, so maybe it's easier to read/understand like:
```
IF NEW.review_response_link_id IS NOT NULL THEN
	   NEW.original_response_id = (SELECT original_response_id FROM review_response where id = NEW.review_response_link_id)
ELSE
          NEW.original_response_id = NEW.id
END IF;
```
Apologies if syntax not exactly right, didn't test it, just a possible suggestion.

Reply:
i think we dont' set application_response_id on lvl above 1

Reply:
agree to both:
* application_response_id to be on all review_responses 

check out the next PR up where template_element_id is added, we can change logic there if that's ok

Reply:
Cool.